includes/Checker/Runtime_Environment_Setup.php

```diff
@@ -100,4 +100,49 @@ public function cleanup() {
 			}
 		}
 	}
+
+	/**
+	 * Checks if the WordPress Environment can be set up for runtime checks.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return bool Returns true if the runtime environment can be set up, false if not.
+	 */
+	public function can_set_up() {
+		global $wp_filesystem;
+
+		require_once ABSPATH . '/wp-admin/includes/upgrade.php';
+
+		if ( ! is_object( $wp_filesystem ) && ! WP_Filesystem() ) {
+			return false;
+		}
+
+		// Check if the object-cache.php file exists.
+		if ( $wp_filesystem->exists( WP_CONTENT_DIR . '/object-cache.php' ) ) {
+			// Check If the object-cache.php file is the Plugin Check version.
+			if ( defined( 'WP_PLUGIN_CHECK_OBJECT_CACHE_DROPIN_VERSION' ) && WP_PLUGIN_CHECK_OBJECT_CACHE_DROPIN_VERSION ) {
+				return true;
+			}
+		} else {
+			// Get the correct Plugin Check directory when run too early.
+			if ( ! defined( 'WP_PLUGIN_CHECK_PLUGIN_DIR_PATH' ) ) {
+				$plugins_dir       = defined( 'WP_PLUGIN_DIR' ) ? WP_PLUGIN_DIR : WP_CONTENT_DIR . '/plugins';
+				$object_cache_copy = $plugins_dir . '/plugin-check/object-cache.copy.php';
+			} else {
+				$object_cache_copy = WP_PLUGIN_CHECK_PLUGIN_DIR_PATH . 'object-cache.copy.php';
+			}
+
+			// If the file does not exist, check if we can place it.
+			$wp_filesystem->copy( $object_cache_copy, WP_CONTENT_DIR . '/object-cache.php' );
+
+			if ( $wp_filesystem->exists( WP_CONTENT_DIR . '/object-cache.php' ) ) { // @phpstan-ignore-line
```

Comment:
Curious, what did PHPStan complain about with this line? Can we exclude the line only for that specific complaint rather than generally? Not familiar, but at least with PHPCS such specific exclusions are possible.

Reply:
@felixarntz PHPStan shows an `If condition is always false` error, and they don't have a specific rule to exclude this complaint. Someone has already opened an issue upstream at https://github.com/phpstan/phpstan/issues/7491.

Until the next release, ignoring the line seems like a better solution in my opinion. cc. @jjgrainger

Reply:
Another option we can remove in `phpstan.neon` like https://gitlab.com/foodsharing-dev/foodsharing/-/blob/master/phpstan.neon#L19

```
-
  message: '#If condition is always true#'
  paths:
     - includes/Checker/Runtime_Environment_Setup.php

```

Reply:
@jjgrainger Okay, then I'd say let's keep the ignore statement as is, but please add a comment (for human readers of the code explaining what is being ignored there and why.

Reply:
Could you please add a comment here to clarify the reason for the ignore statement for people reviewing this code in the future? See https://github.com/10up/plugin-check/pull/173/files#r1210423380.

Reply:
FWIW, `WP_Filesystem_Base::copy` returns `false` if it fails, so we could use that value and not do an additional check to see if the placed file exists. See: https://developer.wordpress.org/reference/classes/wp_filesystem_base/copy/